File: scripts/count_images.py

import os
import re

# Regex to find image paths
# Matches strings starting with /assets/img, ./assets, or absolute URLs, or ending in image extensions
# We look for strings inside quotes
image_pattern = re.compile(r'[\"\']([^\"\']+\.(?:jpg|jpeg|png|webp|svg|gif)(?:[?][^\"\']*)?)[\"\']', re.IGNORECASE)

root_dir = 'src'
unique_images = set()

for subdir, dirs, files in os.walk(root_dir):
    for file in files:
        if file.endswith(('.js', '.jsx')):
            filepath = os.path.join(subdir, file)
            try:
                with open(filepath, 'r', encoding='utf-8') as f:
                    content = f.read()
                    matches = image_pattern.findall(content)
                    for match in matches:
                        if '/assets/' in match or 'http' in match:
                             unique_images.add(match)
            except Exception as e:
                print(f'Error reading {filepath}: {e}')

print(f'Total Unique Images Found: {len(unique_images)}')
for img in sorted(unique_images):
    print(img)

File: scripts/count_images.js

const fs = require('fs');
const path = require('path');

const rootDir = path.join(__dirname, '../src');
const uniqueImages = new Set();

// Recursive function to walk directories
function walkDir(dir) {
    const files = fs.readdirSync(dir);
    files.forEach(file => {
        const filePath = path.join(dir, file);
        const stats = fs.statSync(filePath);
        if (stats.isDirectory()) {
            walkDir(filePath);
        } else if (file.endsWith('.js') || file.endsWith('.jsx')) {
            const content = fs.readFileSync(filePath, 'utf-8');
            // Regex to match image strings
            const regex = /["']([^"']+\.(?:jpg|jpeg|png|webp|svg|gif)(?:[?][^"']*)?)["']/gi;
            let match;
            while ((match = regex.exec(content)) !== null) {
                const imgPath = match[1];
                if (imgPath.includes('/assets/') || imgPath.includes('http')) {
                    uniqueImages.add(imgPath);
                }
            }
        }
    });
}

walkDir(rootDir);

console.log(`Total Unique Images Found: ${uniqueImages.size}`);
const sortedImages = Array.from(uniqueImages).sort();
sortedImages.forEach(img => console.log(img));
